libs/transaction/domain/src/lib/+state/transaction.effects.ts

```diff
@@ -0,0 +1,53 @@
+import { inject, Injectable } from '@angular/core';
+import { Actions, concatLatestFrom, createEffect, ofType } from '@ngrx/effects';
+import { catchError, concatMap, exhaustMap, map, of } from 'rxjs';
+import { TransactionService } from '../infrastructure/transaction.service';
+import {
+  TransactionApiActions,
+  TransactionListActions,
+} from './transaction.actions';
+import { Store } from '@ngrx/store';
+import { userFeature } from '@ngbank/user/store';
+
+@Injectable({ providedIn: 'root' })
+export class TransactionEffects {
+  private readonly transactionService: TransactionService =
+    inject(TransactionService);
+  private readonly actions$: Actions = inject(Actions);
+  private readonly store: Store = inject(Store);
+
+  loadTransactions$ = createEffect(() => {
+    return this.actions$.pipe(
+      ofType(TransactionListActions.opened),
+      exhaustMap(() =>
+        this.transactionService.getTransactions().pipe(
+          map((transactions) =>
+            TransactionApiActions.transactionsLoadedSuccess({ transactions })
+          ),
+          catchError((error) =>
+            of(TransactionApiActions.transactionsLoadedFailure({ error }))
+          )
+        )
+      )
+    );
+  });
+
+  createTransaction$ = createEffect(() => {
+    return this.actions$.pipe(
+      ofType(TransactionListActions.createTransaction),
+      concatLatestFrom(() => this.store.select(userFeature.selectUser)),
+      concatMap(([{ transaction }, user]) =>
+        this.transactionService
+          .createTransaction(transaction, user?.$id ?? '')
```

Comment:
@santoshyadavdev because `user` can be null I have to handle `$id` as optional (see also TransactionService). In practice the whole feature is protected by a guard, so `user` can't be null. Is there a way to get rid of this null handling?

Reply:
Yes you can use user: NonNullable<UserTypes>

Reply:
@santoshyadavdev thanks. I found out that we don't need the user object at this point so I removed it. As of https://appwrite.io/docs/advanced/platform/permissions#default-values the creator of a document will be granted read, update, and delete permissions by default.